discord/commands.go

```diff
@@ -675,6 +698,41 @@ func (bot *Bot) HandleCommand(isAdmin, isPermissioned bool, sett *storage.GuildS
 			log.Println(fmt.Sprintf("\"%s\"", strings.Join(args, " ")))
 			break
 
+		case Map:
+			if len(args[1:]) == 0 {
+				embed := ConstructEmbedForCommand(prefix, cmd, sett)
+				s.ChannelMessageSendEmbed(m.ChannelID, embed)
+			} else {
+				mapVersion := args[len(args)-1]
+
+				var mapName string
+				switch mapVersion {
+				case "simple", "detailed":
+					mapName = strings.Join(args[1:len(args)-1], " ")
+				default:
+					mapName = strings.Join(args[1:], " ")
+					mapVersion = sett.GetMapVersion()
+				}
+				mapItem, err := NewMapItem(mapName)
+				if err != nil {
+					log.Println(err)
+					s.ChannelMessageSend(m.ChannelID, sett.LocalizeMessage(&i18n.Message{
+						ID:    "commands.HandleCommand.Map.notFound",
+						Other: "I don't have a map by that name!",
+					}))
+					break
+				}
+				switch mapVersion {
+				case "simple":
```

Comment:
I think the one suggestion I would have here is to allow `mapVersion=""` to fallthrough to "simple". Reason being is that existing settings for guilds will likely default to having "" as the new MapVersion string when unmarshalled from JSON, because that specific MapVersion field isn't found anywhere.

EDIT: To clarify a bit further. For *new* settings created, I see you did specify "simple" as the default. But for *existing* config for guilds, I think the default value in the struct when unmarshalled would be empty

Reply:
Good note. I didn't consider that case. Is there any functionality in the guild settings to set the value to it's default if not defined on bot startup?

Reply:
Also, one question would be whether that logic exists in the `GetMapVersion` getter in GuildSettings or just as a simple if statement after your line reference to override on empty string.

Reply:
Nah, unfortunately it's actually a bit cumbersome to implement. Because the core problem is that the default JSON unmarshaller (basically a mapping function between JSON -> Go Structs) says "oh, if I don't see this value in the JSON, so I'll just resort to whatever the "null" value of that type is. So 0 for ints/floats, "" for strings, etc. 

It's possible to overwrite the default unmarshaller, but it's pretty painful. There's also approaches like this: https://stackoverflow.com/a/38487668 but they're also kind of hacky. I think in terms of program/control flow, saying "well if no preference was defined, the simple map is probably reasonable", but even that is a bit imperfect, especially for more complicated default values that might crop up in the future.

Reply:
Oh yeah the getter would probably do the trick, good call! It would be safe to always return "simple" if the value is ""